File: src/jsMain/kotlin/Toast.kt
import kotlinx.browser.document
import kotlinx.browser.window
import org.w3c.dom.HTMLDivElement

object Toast {
  private var toastTimeout: Int? = null
  const val short = 3500
  const val long = 6000
  private val el: HTMLDivElement get() = document.getElementById("toast") as HTMLDivElement

  operator fun invoke(message: String, duration: Int = short) {
    if (toastTimeout != null) {
      window.clearTimeout(toastTimeout!!)
      toastClose()
    }
    toastTimeout = window.setTimeout({
      toastClose()
    }, duration)
    el.run {
      this.textContent = message
      this.style.opacity = "1"
    }
  }

  private fun toastClose() {
    el.style.opacity = "0"
    toastTimeout = null
  }
}

File: static/js/toast.js
let toastTimeout = null
let el = document.getElementById('toast')

export function show(text, duration = 3500) {
    if (toastTimeout != null) {
        window.clearTimeout(toastTimeout)
        close()
    }
    toastTimeout = window.setTimeout(() => {
        close()
    }, duration)
    el.textContent = text
    el.style.opacity = '1'
}

export function close() {
    el.style.opacity = '0'
    toastTimeout = null
}
